Add "Using Google Fonts" recipe to Style section (#15911)
gatsbyjs/gatsby: docs/docs/recipes.md

- More on [importing assets into files](/docs/importing-assets-into-files/]
- [Using Typography.js for Google fonts](/docs/typography-js/)

### Using Google Fonts

#### Prerequisites

- A [Gatsby site](/docs/quick-start)
- The [Gatsby CLI](/docs/gatsby-cli/) installed
- Choosing a font package from [the typefaces project](https://github.com/KyleAMathews/typefaces)

#### Directions

1. Run `npm install --save yourchosenfont`, replacing `yourchosenfont` with the name of the font you want to install from [the typefaces project](https://github.com/KyleAMathews/typefaces).

> 2. Add `import "yourchosenfont"` to a layout template, page component, or `gatsby-browser.js`.

Reviewer: I'm sorry I didn't notice this in my earlier review–but I I think a code block in addition to the text would help make this step more scannable. How about something like:

```jsx:title=src/components/layout.js
import "yourchosenfont"
```

3. Once it's imported, you can reference the font in a CSS stylesheet.

```css:title=src/components/layout.css
body {
    font-family: yourchosenfont;
} 
```

> Since you are now hosting your own Google font within your project, there is no need to pull it in when your site loads, increasing your site's speedindex by up to ~300 miliseconds on desktop and 1+ seconds on 3g!

Reviewer: Now that I'm looking at this again, I think this would make a great introductory sentence before the prerequisites. How about something like:

> Hosting your own Google fonts locally within a project means they won't have to be fetched over the network when your site loads...

What do you think?

- More on [importing assets into files](/docs/importing-assets-into-files/]
- [Using Typography.js for Google fonts](/docs/typography-js/)

### Using Google Fonts

#### Prerequisites

- A [Gatsby site](/docs/quick-start)
- The [Gatsby CLI](/docs/gatsby-cli/) installed

> - Choose a font package from [the typefaces project](https://github.com/KyleAMathews/typefaces)

Reviewer: ```suggestion
- Choosing a font package from [the typefaces project](https://github.com/KyleAMathews/typefaces)
```

#### Directions

> 1. Visit [the typefaces project](https://github.com/KyleAMathews/typefaces) and choose the font you'd like.

Reviewer: I think this step can be removed since it's also in the prerequisites.
```suggestion
```

> 2. Run `npm install --save yourchosenfont`, replacing `yourchosenfont` with the name of the font you want to install.

Reviewer: ```suggestion
1. Run `npm install --save yourchosenfont`, replacing `yourchosenfont` with the name of the font you want to install from [the typefaces project](https://github.com/KyleAMathews/typefaces).
```

> 3. Add `require("yourchosenfont")` to your site's entry file.

Reviewer: We typically recommend ES6 imports these days, and there's more than one place to put it:
```suggestion
2. Add `import "yourchosenfont"` to a layout template, page component, or `gatsby-browser.js`.
```

> 4. Once installed, you can add the font either in your CSS stylesheet or inline using `font-family: yourchosenfont`. An inline implementation might look something like this:

Reviewer: I'd recommend showing only one method here, and a global stylesheet is the most common.
```suggestion
3. Once it's imported, you can reference the font in a CSS stylesheet.
```

> ```js:title=src/pages/index.js

Reviewer: ```suggestion
```css:title=src/components/layout.css
```

> <MyComponent style={{font-family: yourchosenfont}} />

Reviewer: ```suggestion
body {
    font-family: yourchosenfont;
} 
```

```

Alternatively, you could use similar code inline on an element within your component as well:

> ```js:title=src/components/MyComponent.js

Reviewer: I think this whole example can drop, since it's more common (and recommended) to use a CSS stylesheet.

Author: Yeah, I didn't include it in the initial commit but went back and did it just expand. But the CSS example does seem more fitting.

import React, { Component } from 'react'

class MyComponent extends Component {
  const googleFontStyle = {
    font-family: yourchosenfont
  }

  render() {
    return <div>
      <h1 style={googleFontStyle}>This is a header using Google Fonts!</h1>
    </div>
  }
}
```

Since you are now hosting your own Google font within your project, there is no need to pull it in when your site loads, increasing your site's speedindex by up to ~300 miliseconds on desktop and 1+ seconds on 3g!

#### Additional Resources

> - [Typography.js](https://www.gatsbyjs.org/docs/typography-js/) - Another option for using Google fonts on a Gatsby site

Reviewer: ```suggestion
- [Typography.js](/docs/typography-js/) - Another option for using Google fonts on a Gatsby site
```
